maidsafe/sn_routing
# Startup tweaks

- Add proptest for DKG
- Remove DKG bounce. The issue it was supposed to fix was not reproducible in tests and the implementation was problematic. For example, it only handled the initial DKG after a node joins, but not the subsequent DKGs. Fixing that was tricky, because it's difficult to tell whether an incoming DKG message for an unknown session should be bounced because the session is only about to be created, or ignored because the session was already discarded. There might be ways to fix this, but it doesn't seem worth the complexity at the moment. We might want to revisit this in a different PR if necessary.
- Add bootstrap message backlog - cache unknown messages during the bootstrap phase, then process them after being bootstrapped.
- Prevent scheduled timeouts from keeping `Stage` alive even after `Routing` is dropped. Results in faster offline detection.
